client-app/tests/unit/controllers/index-test.js

```diff
@@ -43,4 +43,14 @@ module("Unit | Controller | index", function (hooks) {
       "with correct terms"
     );
   });
+
+  test("Creating inline grouping patterns finds the longest matching prefix between selected messages", function (assert) {
+    const controller = this.owner.lookup("controller:index");
+    const messages = ["error foo tomtom", "error foo steaky", "error foo bar"];
+
+    assert.deepEqual(
+      controller.findLongestMatchingPrefix(messages),
+      "error foo "
+    );
+  });
```

Comment:
It might be helpful to add a few tests with special characters, not too many, 2-3 additional strings would suffice.